File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletAudioRecorder/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletAudioRecorderExtension: RufletExtension {
  public init() {}

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "AudioRecorder" ? AudioRecorderService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_audio_recorder/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'audio_recorder.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "AudioRecorder":
        return AudioRecorderService(control: control);
      default:
        return null;
    }
  }
}
